src/redux/actions/genes/loadGeneExpression.js

```diff
@@ -7,20 +7,32 @@ import {
 
 import fetchWork from 'utils/work/fetchWork';
 import getTimeoutForWorkerTask from 'utils/getTimeoutForWorkerTask';
+import { findLoadedGenes } from 'utils/genes';
 
 const loadGeneExpression = (
-  experimentId, genes, componentUuid, useDownsampledExpression = false,
+  experimentId,
+  genes,
+  componentUuid,
 ) => async (dispatch, getState) => {
-  const {
-    loading, matrix, downsampledMatrix,
-  } = getState().genes.expression;
+  const { loading, matrix } = getState().genes.expression.full;
 
   // If other gene expression data is already being loaded, don't dispatch.
   if (loading.length > 0) {
     return null;
   }
+  const { genesToLoad, genesAlreadyLoaded } = findLoadedGenes(matrix, genes);
```

Comment:
Minor refactoring, moved a piece of code to a differet function (to be used by `loadDownsampledGenes` too)